backend/daemon/daemon.go

```diff
@@ -496,3 +513,42 @@ func newNavigationHandler(router *mux.Router) http.Handler {
 		}
 	})
 }
+
+// WithMiddleware generates an grpc option with the given middleware.
+func WithMiddleware(i grpc.UnaryServerInterceptor) grpc.ServerOption {
+	return grpc.UnaryInterceptor(i)
+}
+
+// GetPublicationOnly is a middleware to restrict rpc incoming calls to be getpublication only.
+func GetPublicationOnly(ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler) (interface{}, error) {
+	methodSplitted := strings.Split(info.FullMethod, "/")
+	if len(methodSplitted) < 2 || strings.ToLower(methodSplitted[len(methodSplitted)-1]) != "getpublication" {
+		return nil, fmt.Errorf("Method: %s not allowed. GetPublication only", info.FullMethod)
+	}
+
+	// Calls the handler
+	h, err := handler(ctx, req)
+
+	return h, err
+}
+
+// GwEssentials is a middleware to restrict incoming grpc calls to bare minimum for the gateway to work.
+func GwEssentials(ctx context.Context,
```

Comment:
Consider moving this to the gateway main package, as it seems very tied to it, and only ever used there.